pkg/handler/common/cluster.go

```diff
@@ -185,6 +185,13 @@ func CreateEndpoint(ctx context.Context, projectID string, body apiv1.CreateClus
 		partialCluster.Spec.EnableUserSSHKeyAgent = body.Cluster.Spec.EnableUserSSHKeyAgent
 	}
 
+	partialCluster.Spec.Features = make(map[string]bool)
+	if body.Cluster.Spec.ExternalCloudProvider == nil {
+		partialCluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] = false
```

Comment:
We also have some logic for enabling the `ClusterFeatureExternalCloudProvider` on L198-200. Should we merge this code with the code below to avoid confusion?